archive/dug_6/dug_6_dasharo_configuration_utility_status_update.md

```diff
@@ -0,0 +1,257 @@
+- Customizing the boot logo
+- Setting unique UUIDs or Serial Numbers in SMBIOS tables
+
+???
+
+- Our dedicated tool that enabled customization of Dasharo binaries for users
+- Published around 7 months ago
+- Introduced at Dasharo User Group #4
+- Upon release allowed only for customization of the boot splash logo and
+  managing the Serial Numbers for devices
+
+---
+
+# Reminder: What is DCU?
+
+_The Dasharo Configuration Utility is a tool designed to configure Dasharo firmware binary images._
+
+.left-column50[
+.code-15px[
+```sh
+$ git clone https://github.com/Dasharo/dcu.git
+$ cd dcu
+$ ./dcuc logo coreboot.rom -l bootsplash.bmp
+```
+]
+]
+
+.right-column50[
+.center[<img src="/img/custom_boot.jpg" width="399px">]
+]
+
+???
+
+- In practice it allowed you to do stuff like this :)
+
+---
+
+# What we envisioned
+
+- Custom Configuration Profiles
+- Sane Defaults Verification
+- Data Migration Tools
+- Enterprise Security Features
+- Chipsec and HSI Checks Integration
+- Integration with Commercial Tools like Binarly
+
+???
+
+- Of course this was only the first iteration and we had ambitious plans how we
+  can expand on the idea and bring more functionality to the users
+- #### Custom Configuration Profiles
+  - Creation of custom configuration profiles that can be applied
+    across multiple systems, ensuring consistency in deployment.
+- #### Sane Defaults Verification
+  - Verifying and aligning user configurations with established sane defaults,
+    ensuring optimal security and performance settings for various hardware
+    setups 
+- #### Data Migration Tools
+  - Smooth transition of data and settings from proprietary BIOS to Dasharo
+  firmware.
+- #### Enterprise Security Features
+  - Advanced security features like TPM configuration, secure boot
+  customization, and hardware-based encryption support.
+- #### Integration of tools like Chipsec and Binarly for security audits
+
+---
+
+# Current Status
+
+- ~~Custom Configuration Profiles~~
+- ~~Sane Defaults Verification~~
+- ~~Data Migration Tools~~
+- ~~Enterprise Security Features~~
+- ~~Chipsec and HSI Checks Integration~~
+- ~~Integration with Commercial Tools like Binarly~~
+
+<br>
+
+.center.image-35[![](/img/slacking_off.png)]
+
+???
+
+- And here is what we actually delivered.
+- But that doesn't mean we've been inactive.
+
+---
+
+# The problem
+
+.center[
+<img src="/img/x86_ring_level_arrow.svg" height="550px" style="margin-top:-70px; margin-left:20px">
+]
+
+???
+
+- _SMM is a special operating mode in x86 processors that handles system-wide
+functions like:_
+  - _power management_
+  - _system hardware control_
+  - _firmware settings storage_
+- This separates this funtionality from the userspace protecting it from
+  software attacks
+- Coreboot stores it's firmware settings in the SMMS
+  - Until recently they could be only modified by hand from the UEFI menu
+  - So we made a tool for changing the settings via an util/smmstoretool
```

Comment:
I advise to put some of these on slides.
I am not sure of the ring level graphic here.
It would be better to link to the tools you talk about (like smmstoretool).

Reply:
I am still not sold on the usefulness of this graphic in the context of DCU, but I trust that you will convince me during presentation.

Reply:
Hope I did ;)